homeassistant/components/sensor/__init__.py

```diff
@@ -512,6 +532,13 @@ def unit_of_measurement(self) -> str | None:
         if self._sensor_option_unit_of_measurement:
             return self._sensor_option_unit_of_measurement
 
+        if (
+            self.registry_entry
+            and (sensor_options := self.registry_entry.options.get(DOMAIN))
+            and (display_unit := sensor_options.get("initial_unit_of_measurement"))
```

Comment:
I don't like the word initial_unit_of_measurement here. Shouldn't we set it to `suggested_unit_of_measurement`? It's the one suggested by the entity itself.